frame/staking/src/lib.rs

```diff
@@ -582,36 +640,110 @@ pub struct IndividualExposure<AccountId, Balance: HasCompact> {
 }
 
 /// A snapshot of the stake backing a single validator in the system.
+/// `Limit` is the size limit of `others` bounded by `MaxNominatorRewardedPerValidator`
 #[derive(
-	PartialEq, Eq, PartialOrd, Ord, Clone, Encode, Decode, Default, RuntimeDebug, TypeInfo,
+	PartialEqNoBound,
+	EqNoBound,
+	Encode,
+	Decode,
+	CloneNoBound,
+	DefaultNoBound,
+	RuntimeDebugNoBound,
+	TypeInfo,
+	MaxEncodedLen,
 )]
-pub struct Exposure<AccountId, Balance: HasCompact> {
+#[scale_info(skip_type_params(Limit))]
+#[codec(mel_bound(Limit: Get<u32>, Balance: HasCompact))]
+pub struct Exposure<AccountId, Balance, Limit>
+where
+	AccountId: MaxEncodedLen + Eq + Default + Clone + fmt::Debug,
+	Balance: HasCompact + MaxEncodedLen + Eq + Default + Clone + fmt::Debug,
+	Limit: Get<u32>,
+{
 	/// The total balance backing this validator.
 	#[codec(compact)]
 	pub total: Balance,
 	/// The validator's own stash that is exposed.
 	#[codec(compact)]
 	pub own: Balance,
 	/// The portions of nominators stashes that are exposed.
-	pub others: Vec<IndividualExposure<AccountId, Balance>>,
+	pub others: WeakBoundedVec<IndividualExposure<AccountId, Balance>, Limit>,
+}
+
+impl<AccountId, Balance, Limit> PartialOrd for Exposure<AccountId, Balance, Limit>
```

Comment:
seems like the derive failed and maybe we can fix that for your with frame support custom derives.

Reply:
Indeed, `OrdNoBound` and `PartialOrdNoBound` do not exist, hence why I implemented it manually.

Reply:
Implemented it, if you can please take a look and let me know if it looks good.